vectorstores/alloydb/engine.go

```diff
@@ -50,24 +44,35 @@ func NewPostgresEngine(ctx context.Context, opts ...Option) (*PostgresEngine, er
 }
 
 // createConnection creates a connection pool to the PostgreSQL database.
-func createConnection(ctx context.Context, cfg engineConfig) (*pgxpool.Pool, error) {
-	// Create a new dialer with any options
-	d, err := alloydbconn.NewDialer(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to initialize connection: %w", err)
+func createConnection(ctx context.Context, cfg engineConfig, usingIAMAuth bool) (*pgxpool.Pool, error) {
+	var d *alloydbconn.Dialer
```

Comment:
nit: It would be better for some of these changes to be in a separate PR, as they seem unrelated to ChatMessageHistory